src/synthesizer/python/tests.rs

```diff
@@ -29,6 +34,162 @@ fn test_alexa_org() {
 #[test]
 fn test_condition_ir_not_simple() {
     let condition_ir = ConditionIr::Not(Box::new(ConditionIr::Condition("condition".into())));
-    let result = synthesize_condition_recursive(&condition_ir);
+    let result = synthesize_condition_recursive(&condition_ir, StackType::Stack);
     assert_eq!("not (condition)", result);
 }
+
+#[test]
+fn test_condition_ir_map() {
+    let condition_ir = ConditionIr::Map(
+        "ConditionIrMap".into(),
+        Box::new(ConditionIr::Str("FirstLevelKey".into())),
+        Box::new(ConditionIr::Str("SecondLevelKey".into())),
+    );
+    let result = synthesize_condition_recursive(&condition_ir, StackType::Stack);
+    assert_eq!(
+        "condition_ir_map['FirstLevelKey']['SecondLevelKey']",
+        result
+    );
+}
+
+const SIMPLE_TEMPLATE: &str = r#"{
+    "AWSTemplateFormatVersion": "2010-09-09",
+    "Description": "Test template",
+    "Resources": {
+        "MyBucket": {
+            "Type": "AWS::S3::Bucket",
+            "Properties": {
+                "BucketName": {"Fn::Sub": "${AWS::StackName}-bucket"}
+            }
+        }
+    }
+}"#;
+
+#[test]
+fn test_stack_type_stack_mode() {
+    let cfn: CloudformationParseTree = serde_json::from_str(SIMPLE_TEMPLATE).unwrap();
+    let ir = CloudformationProgramIr::from(cfn, Schema::builtin()).unwrap();
+
+    let mut output = Vec::new();
+    ir.synthesize("python", &mut output, "TestStack", StackType::Stack)
+        .unwrap();
+    let code = String::from_utf8(output).unwrap();
+
+    assert!(
+        code.contains("class TestStack(Stack):"),
+        "Should extend Stack"
+    );
+    assert!(
+        code.contains("super().__init__(scope, construct_id, **kwargs)"),
+        "Should call super with kwargs"
+    );
+    assert!(
+        code.contains("self.stack_name"),
+        "Should use self.stack_name for pseudo-params"
+    );
+}
+
+#[test]
+fn test_stack_type_construct_mode() {
+    let cfn: CloudformationParseTree = serde_json::from_str(SIMPLE_TEMPLATE).unwrap();
+    let ir = CloudformationProgramIr::from(cfn, Schema::builtin()).unwrap();
+
+    let mut output = Vec::new();
+    ir.synthesize("python", &mut output, "TestStack", StackType::Construct)
+        .unwrap();
+    let code = String::from_utf8(output).unwrap();
+
+    assert!(
+        code.contains("class TestStack(Construct):"),
```

Comment:
Its a little confusing for the class itself to be named `TestStack`, but it isn't a stack. 

For construct mode, can we make the class be named `Test`, or `TestConstruct`? For stack mode we should keep `TestStack` to preserve backwards compatibility.